### ISSUE TITLE
Impossible to deploy a GCP Cluster - postgresql password authentication failed, or service account problem

### ISSUE THREAD

**Issue by quematech** [pachyderm/pachyderm#7059]
Hello

I obviously hope I did not miss anything in the documentation, but I'm sadly totally stuck

**What happened?**:

We're following the GCP deployment method described here : 
https://docs.pachyderm.com/latest/deploy-manage/deploy/google_cloud_platform

Everything runs fine, however, when launching the "helm install" method, we have a clear problem.

**Usecase 1** : trying to use a postgresl Databse created following the documentation with Cloud SQL : 
the cloudsql-auth-proxy pod is in CrashLoopBackOff state. Error is : 

`2021/11/04 16:24:07 current FDs rlimit set to 1048576, wanted limit is 8500. Nothing to do here.
2021/11/04 16:24:07 the default Compute Engine service account is not configured with sufficient permissions to access the Cloud SQL API from this VM. Please create a new VM with Cloud SQL access (scope) enabled under "Identity and API access". Alternatively, create a new "service account key" and specify it using the -credential_file parameter`

It could be a mistake from our side. We put in the values.yaml : 

```
cloudsqlAuthProxy:
  enabled: true
  connectionName: X
  serviceAccount: Y
  resources:
    requests:
      memory: "500Mi"
      cpu:    "250m"
```
However, I'm not sure about the serviceAccount value. I took it from 
`gcloud sql instances describe [INSTANCE]`

line : 
`serviceAccountEmailAddress:Y`

**Usecase 2** : trying to use the default postgresql deployed via Helm, using in values.yaml : 

```
postgresql:
  enabled: true
```

When we launch the helm install command, we can see the the pods are created and running, but the postgresql can not authenticate : 

```
2021-11-04 16:30:47.218 GMT [91] LOG:  database system was shut down at 2021-11-04 15:53:01 GMT
2021-11-04 16:30:47.228 GMT [1] LOG:  database system is ready to accept connections
2021-11-04 16:31:10.312 GMT [115] FATAL:  password authentication failed for user "pachyderm"
2021-11-04 16:31:10.312 GMT [115] DETAIL:  Password does not match for user "pachyderm".
        Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
```

**What you expected to happen?**:

For the cluster to start and work 

**How to reproduce it (as minimally and precisely as possible)?**:

Following the GCP installation recreate the same state every time

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:10:43Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.10-gke.301", GitCommit:"17ad7bd6afa01033d7bd3f02ce5de56f940a915d", GitTreeState:"clean", BuildDate:"2021-08-24T05:18:54Z", GoVersion:"go1.15.15b5", Compiler:"gc", Platform:"linux/amd64"}

- Pachyderm CLI and pachd server version (use `pachctl version`):

2.0.0
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):

GKE
- If you deployed with helm, the values you used (`helm get values pachyderm`):
For the second case : 
```
deployTarget: GOOGLE
pachd:
  serviceAccount:
    additionalAnnotations:
      XXXX
  storage:
    google:
      bucket: XXX
      cred: |
        {
          "type": "service_account",
        }
  worker:
    serviceAccount:
      additionalAnnotations:
        XXXXX
postgresql:
  enabled: true
```

- OS (e.g. from /etc/os-release):

Debian

- Others:

**Comment by mindthevirt** [comment]
@quematech thank you for opening up this issue. The current GCP docs are still in draft and will be finalized next week. However, to unblock your setup, you could use the[ quick setup instructions](https://docs.pachyderm.com/latest/deploy-manage/deploy/quickstart/#google) for GCP.

However, before you do so, please make sure to do the following:

1. `helm uninstall pachyderm`
2. `kubectl get pvc`
3. `kubectl delete pvc <name of pvcs from #2>`
4. `kubectl get jobs`
5. `kubectl delete jobs <name of the jobs from #4>`

Once you've cleaned up your cluster with the above steps, the quick start guide for GCP should bring up a cluster up quickly.

**Comment by quematech** [comment]
Hello

Thank you for your quick answer, it looks like it's working :)

Just a small detail : in the linked documentation, you ask for the service account json to be inserted in the values.yaml. However, documentation does not explain what authorizations the service account should have. I tried with the storage authorization, and it looks like it works, but I'm not sure if this was needed or not.

Best regards

**Comment by quematech** [comment]
Hmm, I was in a hurry, it does not work, but a different error happens (certificate signed by unknown authority). I will create a new issue in order to keep things clean.
